crates/wascc-provider/src/states/container/waiting.rs

```diff
@@ -0,0 +1,272 @@
+                            ),
+                        )
+                    }
+                }
+            } else {
+                vec![]
+            };
+
+        debug!("Starting container {} on thread", container.name());
+
+        let module_data = {
+            let mut run_context = state.run_context.write().await;
+            match run_context.modules.remove(container.name()) {
+                Some(module) => module,
+                None => {
+                    return Transition::next(
+                        self,
+                        Terminated::new(
+                            format!(
+                                "FATAL ERROR: module map not properly populated ({}/{})",
+                                state.pod.name(),
+                                container.name(),
+                            ),
+                            true,
+                        ),
+                    )
+                }
+            }
+        };
+
+        match tokio::task::spawn_blocking(move || {
+            wascc_run(
+                host,
+                module_data,
+                env,
+                volume_bindings,
+                &log_path,
+                port_assigned,
+            )
+        })
+        .await
+        {
+            Ok(Ok(container_handle)) => {
+                let pod_key = PodKey::from(&state.pod);
+                {
+                    let provider_state = shared.write().await;
+                    let mut handles_writer = provider_state.handles.write().await;
+                    let pod_handle = handles_writer
+                        .entry(pod_key)
+                        .or_insert_with(|| PodHandle::new(HashMap::new(), state.pod.clone(), None));
+                    pod_handle
+                        .insert_container_handle(state.container_key.clone(), container_handle)
+                        .await;
+                }
+            }
+            Ok(Err(e)) => {
+                return Transition::next(
+                    self,
+                    Terminated::new(
+                        format!(
+                            "Pod {} container {} failed to start wascc actor: {:?}",
+                            state.pod.name(),
+                            container.name(),
+                            e
+                        ),
+                        true,
+                    ),
+                )
+            }
+            Err(e) => {
+                return Transition::next(
+                    self,
+                    Terminated::new(
+                        format!(
+                            "Pod {} container {} failed to start wascc actor: {:?}",
+                            state.pod.name(),
+                            container.name(),
+                            e
+                        ),
+                        true,
+                    ),
+                )
+            }
+        }
+
+        Transition::next(self, Running)
+    }
```

Comment:
Unless I've missed something, this is a 160-line method with up to 7 levels of indentation in places. Maybe I am dumb but I find that really hard to keep straight in my mind.  I assume that this is probably code that already exists and we are just moving it into a state, so I can see that we might not want to change it as part of this PR, but it should be high on some list for refactoring!

Reply:
Yep, I essentially just pulled in `start_container` and included any necessary context from `Pod::starting`. I think these should be broken up in future refactoring.